learning_journal/tests.py

```
@@ -1,65 +1,249 @@
+    """Test the detail view."""
+    from .views.default import detail
+    dummy_request.method = "POST"
+    dummy_request.POST["title"] = "This is a Title"
+    dummy_request.POST["body"] = "And this is the body. TEST!"
+    dummy_request.matchdict['id'] = '1'
+    result = detail(dummy_request)
+    entry = dummy_request.dbsession.query(Entry).get(1)
+    assert result['entry'] == entry
+
+
+def test_detail_page_returns_empty_when_empty(dummy_request):
+    """Test that the detail view returns no objects when database empty."""
+    from learning_journal.views.default import detail
+    req = dummy_request
+    req.matchdict = {"id": "1"}
+    result = detail(req)
+    assert result["entry"] is None
+
+
+def test_edit_page_updates_db(dummy_request, add_models):
+    """Test that the edit view edits entries in the database."""
+    from learning_journal.views.default import edit
+    req = dummy_request
+    req.matchdict = {"id": "3"}
+    req.method = "POST"
+    req.POST["title"] = "new thing"
+    req.POST["creation_date"] = datetime.datetime(2017, 2, 24, 0, 0)
+    req.POST["body"] = "and more new things"
+    try:
+        edit(req)
+    except:
+        new_title = dummy_request.dbsession.query(Entry).get(3).title
+        assert new_title == "new thing"
+
+
+def test_write_view_updates_db(dummy_request, add_models):
+    """Test that the write view adds entries to the database."""
+    from learning_journal.views.default import write
+    row_count_before_post = dummy_request.dbsession.query(Entry).count()
+    req = dummy_request
+    req.method = "POST"
+    req.POST["title"] = "a new post"
+    req.POST["title1"] = "a new subtitle"
+    req.POST["creation_date"] = datetime.datetime(2016, 12, 20, 0, 0)
+    req.POST["body"] = "a new body"
+    try:
+        write(req)
+    except:
+        row_count_after_post = dummy_request.dbsession.query(Entry).count()
+        assert row_count_after_post == row_count_before_post + 1
```

Again, why is there a bare except?